src/web/components/App.tsx

```diff
@@ -327,7 +336,49 @@ export const App = ({ CAPI, NAV, ophanRecord }: Props) => {
 				pubData,
 			});
 		}
-	}, [countryCode, CAPI.config.switches.consentManagement, pageViewId]);
+	}, [
+		countryCode,
+		CAPI.config.switches.consentManagement,
+		pageViewId,
+		browserId,
+	]);
+
+	// ************************
+	// *      Commercial      *
+	// ************************
+	const [sentCommercialMetrics, setSentCommercialMetrics] = useState<boolean>(
+		false,
+	);
+	const visibilityState = useDocumentVisibilityState();
+	useEffect(() => {
+		if (!window.guardian.config.switches.commercialMetrics) return;
+		if (!window.guardian.config.ophan) return;
+		log('commercial', 'document.visibilityState:', visibilityState);
+		if (visibilityState !== 'hidden') return;
+		if (sentCommercialMetrics === true) return;
+
+		const shouldForceMetrics = ABTestAPI.allRunnableTests(
+			tests,
+		).some((test) =>
+			[commercialPartner].map((t) => t.id).includes(test.id),
```

Comment:
Is `[commercialPartner]` the array of AB tests for which we want to force metrics? If so could it maybe be declared as a constant at the top with the other constants?

Reply:
I was wondering about that. Let's make it explicit, and then we can also type it to an array of `ABTest`